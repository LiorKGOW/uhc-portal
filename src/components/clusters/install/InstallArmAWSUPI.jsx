import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { PageSection } from '@patternfly/react-core';

import Breadcrumbs from '../../common/Breadcrumbs';
import { tollboothActions } from '../../../redux/actions';
import { scrollToTop } from '../../../common/helpers';
import instructionsMapping from './instructions/instructionsMapping';
import OCPInstructions from './instructions/OCPInstructions';
import PageTitle from '../../common/PageTitle';

export class InstallArmAWSUPI extends Component {
  componentDidMount() {
    scrollToTop();
    document.title =
      'Install OpenShift 4 | Red Hat OpenShift Cluster Manager | AWS User-Provisioned ARM Infrastructure';

    const { dispatch } = this.props;
    dispatch(tollboothActions.createAuthToken());
  }

  render() {
    const { token } = this.props;
    const breadcrumbs = (
      <Breadcrumbs
        path={[
          { label: 'Clusters' },
          { label: 'Create', path: '/create' },
          { label: 'Amazon Web Services (ARM)', path: '/install/aws/arm' },
          { label: 'User-provisioned infrastructure' },
        ]}
      />
    );

    return (
      <>
        <PageTitle title={instructionsMapping.aws.arm.upi.title} breadcrumbs={breadcrumbs} />
        <PageSection>
          <OCPInstructions
            token={token}
            cloudProviderID="aws"
            isUPI
            {...instructionsMapping.aws.arm.upi}
          />
        </PageSection>
      </>
    );
  }
}

InstallArmAWSUPI.propTypes = {
  token: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({ token: state.tollbooth.token });

export default connect(mapStateToProps)(InstallArmAWSUPI);
