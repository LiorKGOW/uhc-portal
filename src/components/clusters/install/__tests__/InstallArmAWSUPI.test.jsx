import React from 'react';
import { shallow } from 'enzyme';

import { InstallArmAWSUPI } from '../InstallArmAWSUPI';

describe('InstallArmAWSUPI', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<InstallArmAWSUPI token={{}} dispatch={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
