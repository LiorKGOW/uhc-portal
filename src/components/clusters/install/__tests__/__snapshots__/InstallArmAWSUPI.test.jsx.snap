// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallArmAWSUPI renders correctly 1`] = `
<Fragment>
  <PageTitle
    breadcrumbs={
      <Breadcrumbs
        path={
          Array [
            Object {
              "label": "Clusters",
            },
            Object {
              "label": "Create",
              "path": "/create",
            },
            Object {
              "label": "Amazon Web Services (ARM)",
              "path": "/install/aws/arm",
            },
            Object {
              "label": "User-provisioned infrastructure",
            },
          ]
        }
      />
    }
    title="Install OpenShift on AWS with user-provisioned ARM infrastructure"
  />
  <PageSection>
    <OCPInstructions
      channel="stable"
      cloudProviderID="aws"
      docURL="https://docs.openshift.com/container-platform/4.11/installing/installing_aws/installing-aws-user-infra.html"
      installer="aarch64-openshift-install"
      isBMIPI={false}
      isUPI={true}
      preReleasePageLink="/install/arm/pre-release"
      title="Install OpenShift on AWS with user-provisioned ARM infrastructure"
      token={Object {}}
    />
  </PageSection>
</Fragment>
`;
